src/test/java/bio/terra/workspace/service/workspace/flight/CreateGoogleContextRBSFlightTest.java

```diff
@@ -1,33 +1,213 @@
+    inputs.put(WorkspaceFlightMapKeys.WORKSPACE_ID, workspaceId);
+    inputs.put(WorkspaceFlightMapKeys.BILLING_ACCOUNT_ID, billingAccountId);
+    inputs.put(JobMapKeys.AUTH_USER_INFO.getKeyName(), userReq);
+    return inputs;
+  }
+
+  private void assertServiceApisEnabled(Project project, List<String> enabledApis)
+      throws Exception {
+    List<String> serviceNames =
+        enabledApis.stream()
+            .map(
+                apiName ->
+                    String.format("projects/%d/services/%s", project.getProjectNumber(), apiName))
+            .collect(Collectors.toList());
+    ListServicesResponse servicesList =
+        crl.getServiceUsageCow()
+            .services()
+            .list("projects/" + project.getProjectId())
+            .setFilter("state:ENABLED")
+            .execute();
+    assertThat(
+        servicesList.getServices().stream()
+            .map(GoogleApiServiceusageV1Service::getName)
+            .collect(Collectors.toList()),
+        Matchers.hasItems(serviceNames.toArray()));
+  }
+
+  /** Asserts that Sam groups are granted their appropriate IAM roles on a GCP project. */
+  private void assertPolicyGroupsSynced(UUID workspaceId, Project project) throws Exception {
+    Map<IamRole, String> roleToSamGroup =
+        Arrays.stream(IamRole.values())
+            .collect(
+                Collectors.toMap(
+                    Function.identity(),
+                    role ->
+                        "group:"
+                            + samService.syncWorkspacePolicy(
+                                workspaceId, role, userAccessUtils.defaultUserAuthRequest())));
+    Policy currentPolicy =
+        crl.getCloudResourceManagerCow()
+            .projects()
+            .getIamPolicy(project.getProjectId(), new GetIamPolicyRequest())
+            .execute();
+    for (IamRole role : IamRole.values()) {
+      assertRoleBindingsInPolicy(role, roleToSamGroup.get(role), currentPolicy);
+    }
+  }
+
+  /**
+   * Validate that a GCP policy contains expected role bindings.
+   *
+   * @param role An IAM role. Maps to expected GCP roles in CloudSyncRoleMapping.
+   * @param groupEmail The group we expect roles to be bound to.
+   * @param gcpPolicy The GCP policy we're checking for role bindings.
+   */
+  private void assertRoleBindingsInPolicy(IamRole role, String groupEmail, Policy gcpPolicy) {
+    List<String> expectedGcpRoleList = CloudSyncRoleMapping.cloudSyncRoleMap.get(role);
+    List<Binding> actualGcpBindingList = gcpPolicy.getBindings();
+    List<String> actualGcpRoleList =
+        actualGcpBindingList.stream().map(Binding::getRole).collect(Collectors.toList());
+    for (String gcpRole : expectedGcpRoleList) {
+      assertTrue(actualGcpRoleList.contains(gcpRole));
+      assertTrue(
+          actualGcpBindingList
+              .get(actualGcpRoleList.indexOf(gcpRole))
+              .getMembers()
+              .contains(groupEmail));
+    }
+  }
```

Comment:
should you also check that the actual list doesn't have extra roles or members that it shouldn't?